pre_commit_hooks/check_google_doc_style.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+"""Checks doc style against https://developers.google.com/style."""
+
+__copyright__ = """
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import re
+import sys
+
+_IGNORE_START = "<!-- google-doc-style-ignore -->"
+_IGNORE_STOP = "<!-- google-doc-style-resume -->"
+
+# These are from https://developers.google.com/style/word-list.
+_REPLACERS = (
+    ("blacklist", "blocklist"),
+    ("cons", "disadvantages"),
+    ("e.g.", "for example"),
+    ("file name", "filename"),
+    ("filesystem", "file system"),
+    ("flag", "option"),
+    ("flags", "options"),
+    ("for instance", "for example"),
+    ("i.e.", "that is"),
+    ("pros", "advantages"),
+    ("repo", "repository"),
+    ("repos", "repositories"),
+    ("tl;dr", "to summarize"),
+    ("via", "by way of"),
+    ("vice versa", "the other way around"),
+    ("vs.", "versus"),
+    ("whitelist", "allowlist"),
+)
+
+
+def _parse_args(argv=None):
+    """Parses command-line arguments and flags."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "paths",
+        metavar="PATH",
+        nargs="+",
+        help="One or more paths of files to check.",
+    )
+    return parser.parse_args(args=argv)
+
+
+def build_replacers():
+    """Builds replacement regex objects."""
+    for capitalize in (True, False):
+        for before, after in _REPLACERS:
+            if capitalize:
+                before = before.capitalize()
+                after = after.capitalize()
```

Comment:
Why not use a case-insensitive regex?

Notably, this won't handle `TL;DR` or `I.E.`...

Reply:
Consider the current casing results:
"i.e." -> "that is", "cons" -> "disadvantages"
"I.e." -> "That is", "Cons" -> "Disadvantages"

But, for case-insensitive:
"I.E." -> "THAT IS", "CONS" -> "DISADVANTAGES"?
"i.E." -> ???, "cOns" -> ???

So basically, I wanted to be as accurate as possible. If you think it's worth it, could do a case-insensitive pass after both case-sensitive passes and maybe hand out all-uppercase replacements unconditionally. My leaning is to wait and see on that, though.

Reply:
Actually, went ahead and did the ignore case for capitalize, avoiding special-casing all-caps

Reply:
Hrm, your point about the transformation is interesting. I'm not sure either of these is better or worse given that, so I'm OK with this or with the previous version, whichever is giving better results in practice.